src/opnsense/mvc/app/models/OPNsense/IPsec/IPsec.xml

```diff
@@ -0,0 +1,31 @@
+<model>
+    <mount>//OPNsense/IPsec</mount>
+    <description>
+        OPNsense IPsec
+    </description>
+    <items>
+        <keyPairs>
+            <keyPair type="ArrayField">
+                <name type="TextField">
+                    <Required>Y</Required>
+                </name>
+                <keyType type="OptionField">
+                    <Required>Y</Required>
+                    <default>rsa</default>
+                    <OptionValues>
+                        <rsa>RSA</rsa>
+                    </OptionValues>
+                </keyType>
+                <publicKey type="TextField">
+                    <Required>Y</Required>
+                </publicKey>
+                <privateKey type="TextField">
+                    <Required>N</Required>
+                </privateKey>
+
+                <_keySize type="IntegerField"/>
+                <_keyFingerprint type="TextField"/>
```

Comment:
Please no, underscores here look awful.

Reply:
I have used underscores as these fields are used for caching/storing the key size and fingerprint and should not be touched manually, which is why the UI also does not offer any method for directly editing these properties.

Any other naming suggestion for this purpose?

Reply:
I just made them looking like regular fields in my code…

If you would download the list via the API for some custom application, you would not care if the server generated it or if it comes from storage - you are just interested in the content.

This may cause some problems because some methods in PHP have special names and also start with an underscore - for example `__constructor`.

Reply:
Thank you for your reply, I didn't think about the use case of querying these values through the API - were underscores would indeed be fairly awkward. I have updated the field names to `keySize` and `keyFingerprint`.